### PR TITLE
[FFM-9621] - Avoid sending empty metric payloads

### PR DESCRIPTION
**Pull request by andybharness**
[FFM-9621] - Avoid sending empty metric payloads

What
Adds a check to make sure metrics data is not empty before sending it.

Why
We're seeing empty metrics payloads arriving at the BE we should avoid wasting network bandwidth.

Testing
Manual

[FFM-9621]: https://harness.atlassian.net/browse/FFM-9621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ribeirophillipe — APPROVED** [review]

File: src/metrics.ts [hunk 1]
```diff
@@ -188,7 +188,8 @@ export const MetricsProcessor = (
     }
 
     const metrics: Metrics = _summarize();
-    if (metrics) {
+
+    if (metrics && metrics.metricsData.length + metrics.targetData.length > 0) {
```
Comment (knagurski) [inline src/metrics.ts:192]: Super minor and works the same way, but you can treat these as `truthy`/`falsey`.
```suggestion
    if (metrics && metrics.metricsData.length && metrics.targetData.length) {
```

Comment (knagurski) [reply src/metrics.ts:192]: That is unless you do want to send if `metricsData` is empty, but `targetData` isn't. That that case, your could `||` them
```suggestion
    if (metrics && (metrics.metricsData.length || metrics.targetData.length)) {
```

**Comment by knagurski** [comment]
Do you need to bump the version as well for a release?

**Comment by andybharness** [comment]
I've only bumped the minor since there's no real API changes - let me double check if those files were missed

**Comment by andybharness** [comment]
@knagurski i had been working on a older copy of the git tree and rev'ed the version to 1.3.4, turns out that version had been released recently. I've fixed the version now to be 1.3.5

**Review by knagurski — APPROVED** [review]
Nice and simple 👍🏻

**Pull request merged**